src/amd/compiler/aco_lower_bool_phis.cpp

```diff
@@ -122,22 +123,24 @@ Temp write_ssa(Program *program, Block *block, ssa_state *state, unsigned previo
 
 void insert_before_branch(Block *block, aco_ptr<Instruction> instr)
 {
-   int end = block->instructions.size() - 1;
-   if (block->instructions[end]->format == Format::PSEUDO_BRANCH)
-      block->instructions.emplace(std::prev(block->instructions.end()), std::move(instr));
+   if (block->instructions.back()->format == Format::PSEUDO_BRANCH)
+      block->instructions.insert(std::prev(block->instructions.end()), std::move(instr));
    else
-      block->instructions.emplace_back(std::move(instr));
+      block->instructions.insert(block->instructions.end(), std::move(instr));
 }
 
 void insert_before_logical_end(Block *block, aco_ptr<Instruction> instr)
 {
-   for (int i = block->instructions.size() - 1; i >= 0; --i) {
-      if (block->instructions[i]->opcode == aco_opcode::p_logical_end) {
-         block->instructions.emplace(std::next(block->instructions.begin(), i), std::move(instr));
-         return;
-      }
+   auto IsLogicalEnd = [] (const aco_ptr<Instruction>& instr) -> bool {
+      return instr->opcode == aco_opcode::p_logical_end;
+   };
+   auto it = std::find_if(block->instructions.crbegin(), block->instructions.crend(), IsLogicalEnd);
+
+   if (it == block->instructions.crend()) {
+      insert_before_branch(block, std::move(instr));
+   } else {
+      block->instructions.insert(it.base(), std::move(instr));
    }
```

Comment:
Can the braces around the IF be removed? This is the style I usually use and I think the rest of ACO uses it too.